__tests__/alarm.ts

```diff
@@ -316,9 +316,76 @@ describe('Timer tests', () => {
         }
 
         alarm.set(task, delay0, schedule);
-
      
     });
 
 
+    /**
+     * Test with immmidiateRef when 0 < delay < 25ms
+     */
+
+     it('Our own: sync, wait 10 miliseconds, 20 samples\n\n', function(done) {
+        var i = 0;
+        var j = 0;
+        var numSamples = 20;
+        var doneCount = 0;
+        var errors:number[] = [];
+        var minError = 1000000000;
+        var maxError = 0;
+        var avgError = 0;
+        
+        var hiResDif = process.hrtime(alarm.hiResStart);
+        var delay10Msec = TimeValue.msec(10);
+
+
+        var task = function(){
+            var count = 0;
+            for(i=0;i<1000000;i++){
+                count++;
+            }; 
+        };
+        
+        var schedule = function(tv: TimeValue) {
+            var tuple = tv.toTimeTuple()
+            var waitTime = (tuple[0] * 1000000000) + tuple[1];
+            console.log('\t\t - Sample #' + (doneCount+1));
+            console.log('\t\t\t - Expected wait: 0 seconds');
+            console.log('\t\t\t - Actual wait: ' + waitTime/1000000000 + ' seconds');
+            var error = (((waitTime) / (100000000)) * 100);
+            console.log('\t\t\t - Error: ' + error + '%');
+            errors.push(error);
+            var waitedLongEnough = (waitTime >= 0);
+            expect(waitedLongEnough).toBeTruthy();
+            
+            doneCount++;
+            
+            if(doneCount == numSamples) {
+                for(i=0;i<numSamples;i++){
+                    if(errors[i] < minError){
+                        minError = errors[i];
+                    }
+                    
+                    if (errors[i] > maxError){
+                        maxError = errors[i];
+                    }
+                    
+                    avgError += errors[i];
+                }
+                avgError = avgError / numSamples;
+                console.log('\t\t - Min. Error: ' + minError + '%');
+                console.log('\t\t - Max. Error: ' + maxError + '%');
+                console.log('\t\t - Avg. Error: ' + avgError + '%');
+                done();
+            } else {
+                alarm.set(task, delay10Msec, schedule);        
+            }
```

Comment:
Could we factor this out as a separate function? It looks like this part of the code is repeated multiple times.

Reply:
I try to separate that function out but failed.
I think it is difficult because of the following reason:
The schedule function uses external variables(doneCount, numSamples, etc..), so one way to separate the function is
by replacing these variables with additional arguments.
(For example, **schedule(tv: TimeValue, doneCount: Number, numSamples: Number)** )
However, the schedule function should have a TimeValue argument only, according to line 576 of core/time.ts.
So additional arguments are invalid.

I think there are other ways to separate the function, but couldn't find it.
If I find the way, I will apply it.

Reply:
Ah, I see. No worries! Then I think it's fine to leave it as is now. Thanks for checking!

Reply:
You're welcome! Thank you for your attention to this code!